conda/cli/main_env_create.py

```diff
@@ -155,43 +156,47 @@ def execute(args: Namespace, parser: ArgumentParser) -> int:
         pkg_specs = env.dependencies.get(installer_type, [])
         pkg_specs.extend(args_packages)
 
-        solved_env = installer.dry_run(pkg_specs, args, env)
-        if args.json:
-            print(json.dumps(solved_env.to_dict(), indent=2))
-        else:
-            print(solved_env.to_yaml(), end="")
+        solved_packages = installer.dry_run(pkg_specs, context)
```

Comment:
I'm not sure about this method name. It's more like `solve_records` (inspired by `Solver.solve_for_final_state()`), so we are adding a layer in the middle just when we deal some files, but in `install.py` we call the solver directly 🤔 There's something about that inconsistency that I don't like but I don't have a solution right now.

And yea, as Jannis said, we use the global `context` or pass the necessary bits via parameters.